# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webaudio/MediaStreamAudioSourceNode.h
# Commit: c54cb3c10992
# Full Hash: c54cb3c109922bc15fb6d0ca67e5f2a05980c1a5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 2
# Description:
#   Bug 1493613 - Move MediaStream control from DOMMediaStream to MediaStreamTrack. r=padenot
#   
#   This is inherently large, because modifying these bits of DOMMediaStream and
#   MediaStreamTrack affects all consumers and producers of all DOMMediaStreams and
#   MediaStreamTracks.
# ==============================================================================

diff -r 0b03dd9d20ac -r c54cb3c10992 dom/media/webaudio/MediaStreamAudioSourceNode.h
--- a/dom/media/webaudio/MediaStreamAudioSourceNode.h	Wed Jul 31 09:35:06 2019 +0000
+++ b/dom/media/webaudio/MediaStreamAudioSourceNode.h	Wed Jul 31 07:58:17 2019 +0000
@@ -8,8 +8,9 @@
 #define MediaStreamAudioSourceNode_h_
 
 #include "AudioNode.h"
+#include "AudioNodeEngine.h"
 #include "DOMMediaStream.h"
-#include "AudioNodeEngine.h"
+#include "PrincipalChangeObserver.h"
 
 namespace mozilla {
 
@@ -77,7 +78,7 @@
   size_t SizeOfIncludingThis(MallocSizeOf aMallocSizeOf) const override;
 
   // Attaches to aTrack so that its audio content will be used as input.
-  void AttachToTrack(const RefPtr<MediaStreamTrack>& aTrack);
+  void AttachToTrack(const RefPtr<MediaStreamTrack>& aTrack, ErrorResult& aRv);
 
   // Detaches from the currently attached track if there is one.
   void DetachFromTrack();